delayrecorder.py

```diff
@@ -1,87 +1,175 @@
 import csv
+import numpy as np
+import matplotlib.pyplot as plt
+
+class Stats(object):
+    def __init__(self, num_measurements, min, max, mean, median, std):
+        self.num_measurements = num_measurements
+        self.min = min
+        self.max = max
+        self.mean = mean
+        self.median = median
+        self.std = std
+
+def sigint_handler(signal, frame):
+    print("Caught KeyboardInterrupt, exiting.")
+    sys.exit(0)
+
+def get_measurements_serial(num_measurements, quietMode):
+    # Search for Arduino
+    devs = serial.tools.list_ports.comports()
+    found = 0
+
+    for dev in devs:
+        if dev.manufacturer is not None:
+            if "Arduino" in dev.manufacturer:
+                print("Found Arduino at " + dev[0])
+                ser = serial.Serial(dev[0], 115200,timeout=5)
+                found = 1
+
+    if found == 0:
+        print("Did not find Arduino on any serial port. Is it connected?")
+        sys.exit()
+
+    print("Collecting " + str(num_measurements) + " measurements from the Arduino")
+    if quietMode:
+        print("Running in quiet mode, won't print the measurements to the terminal")
+
+    # Read messages from Arduino
+    timeout = time.time() + 0.01
+    while True:
+        a = ser.readline().decode()
+        if time.time() > timeout:
+            break
+
+    measurements = []
+    i = 0
+    overallrounds = 0
+    initmessage = 0
+
+    while (i < num_measurements):
+        overallrounds = overallrounds + 1
+        a = ser.readline().decode()
+        if "." in a:
+            initmessage = 1
+            i = i + 1
+            a = a.replace("\n","")
+            a = a.replace("\r","")
+            measurements.append(float(a))
+            if not quietMode:
+                print("G2G Delay trial " + str(i) + "/" + str(num_measurements) + ": " + a + " ms")
+            else:
+                print("G2G Delay trial " + str(i) + "/" + str(num_measurements), end='\r')
+        else:
+            if overallrounds > 0 and initmessage == 1:
+                print("Did not receive msmt data from the Arduino for another 5 seconds. Is the phototransistor still sensing the LED?")
+            else:
+                print("Did not receive msmt data from the Arduino for 5 seconds. \n Is the phototransistor sensing the LED on the screen? \n Is the correct side of the PT pointing towards the screen (the flat side with the knob on it)? \n Is the screen brightness high enough (max recommended)?")
+                initmessage = 1
+
+    return measurements
+
+def write_measurements_csv(filename, measurements, stats):
+    with open(filename, 'w') as f:
+        writer = csv.writer(f)
+        writer.writerow(measurements)
+        header = ['Samples', 'Min', 'Max', 'Mean', 'Median', 'Std']
+        writer.writerow(header)
+        stats_list = [stats.num_measurements, stats.min, stats.max, stats.mean, stats.median, stats.std]
+        writer.writerow(stats_list)
```

Comment:
I find it a bit weird that we're writing msmts before the headers. How about

```suggestion
        header = ['Samples', 'Min', 'Max', 'Mean', 'Median', 'StdDev']
        writer.writerow(header)
        writer.writerow([stats.num_measurements, stats.min, stats.max, stats.mean, stats.median, stats.std])
        writer.writerow(['Raw measurements'])
        writer.writerow(measurements)
```